packages/host/app/components/operator-mode-overlays.gts

```diff
@@ -0,0 +1,114 @@
+import Component from '@glimmer/component';
+import { registerDestructor } from '@ember/destroyable';
+
+import { tracked } from '@glimmer/tracking';
+import { fn } from '@ember/helper';
+import { TrackedArray } from 'tracked-built-ins';
+import { Card } from 'https://cardstack.com/base/card-api';
+import { on } from '@ember/modifier';
+import { schedule } from '@ember/runloop';
+import { task, timeout } from 'ember-concurrency';
+import config from '@cardstack/host/config/environment';
+
+interface OverlayButton {
+  x: number;
+  y: number;
+  model: Card;
+  embeddedCardElement: HTMLElement;
+}
+
+interface Signature {
+  Args: {
+    addToStack: (card: Card) => void;
+  };
+}
+
+export default class OperatorModeOverlays extends Component<Signature> {
+  <template>
+    {{#each this.overlayButtons as |overlayButton|}}
+      <button
+        {{on 'click' (fn this.args.addToStack overlayButton.model)}}
+        style='top: {{overlayButton.y}}px; left: {{overlayButton.x}}px; position: absolute; z-index: 2; border: none'
+        data-test-cardstack-operator-mode-overlay-button
+      >
+        Open
+      </button>
+    {{/each}}
+  </template>
+
+  @tracked overlayButtons = new TrackedArray<OverlayButton>([]);
+
+  constructor(owner: unknown, args: any) {
+    super(owner, args);
+
+    (globalThis as any)._CARDSTACK_OPERATOR_MODE_OVERLAYS_COMPONENT = this;
+
+    if (config.environment != 'test') {
+      this.refreshOverlayButtonCoordinates.perform();
+    }
+
+    registerDestructor(this, () => {
+      delete (globalThis as any)._CARDSTACK_OPERATOR_MODE_OVERLAYS_COMPONENT;
+      this.refreshOverlayButtonCoordinates.cancelAll();
+    });
+  }
+
+  registerElement(embeddedCardElement: HTMLElement, model: Card) {
+    schedule('afterRender', () => {
+      let { x, y } =
+        this.calculateOverlayedButtonCoordinates(embeddedCardElement);
+
+      this.overlayButtons.push({
+        x,
+        y,
+        model,
+        embeddedCardElement,
+      });
+    });
+  }
+
+  calculateOverlayedButtonCoordinates(parentElement: HTMLElement) {
+    let rect = parentElement.getBoundingClientRect();
+    return {
+      x: rect.x + rect.width - 60,
+      y: rect.y + 10,
+    };
+  }
+
+  refreshOverlayButtonCoordinates = task(async () => {
+    if (this.overlayButtons.length != 0) {
+      let newOverlayButtons = this.overlayButtons.map((overlayButton) => {
+        let { x, y } = this.calculateOverlayedButtonCoordinates(
+          overlayButton.embeddedCardElement
+        );
+
+        return {
+          x,
+          y,
+          model: overlayButton.model,
+          embeddedCardElement: overlayButton.embeddedCardElement,
+        };
+      });
+
+      if (
+        JSON.stringify(newOverlayButtons) != JSON.stringify(this.overlayButtons) // stringifying is an easy way to compare objects deeply
```

Comment:
you might have some `[Object Object]` values in that string actually since there are card instances and HTMLElements properties in those objects--is that ok?

Reply:
Should be ok in this case but it really does sound unreliable. I improved this check in https://github.com/cardstack/boxel/pull/326/commits/a0b9f086c1cee1547f1bb309691bab4c15341b0f